src/rpdk/core/fragment/generator.py

```diff
@@ -0,0 +1,347 @@
+"""
+This class has two responsibilities:
+1. generating a sample template fragment so the user has some initial
+file fragment as an example.
+The method "generate_sample_fragment" will be called as part of the init command.
+2. generating schema for provided template fragments.
+The method "generate_schema" will be called right before submission.
+"""
+import json
+import logging
+import os
+from pathlib import Path
+
+import yaml
+
+from rpdk.core.data_loaders import resource_json
+from rpdk.core.exceptions import FragmentValidationError
+
+LOG = logging.getLogger(__name__)
+FRAGMENT_DIR = "fragments"
+SAMPLE_FRAGMENT_OUTPUT = "sample.json"
+SCHEMA_NAME = "schema.json"
+SAMPLE_FRAGMENT = "../data/examples/module/sample.json"
+ALLOWED_EXTENSIONS = {".json", ".yaml", ".yml"}
+RESOURCE_LIMIT = 500
+OUTPUT_LIMIT = 200
+MAPPING_LIMIT = 200
+MAPPING_ATTRIBUTE_LIMIT = 200
+TEMPLATE_FILE_SIZE_IN_BYTES_LIMIT = 1500000
```

Comment:
Isn't this over [CloudFormation's template size limit](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/cloudformation-limits.html)?

Reply:
Also think a lot of this validation this isn't specific to modules can be removed since [`cfn-lint` is being called now](https://github.com/aws-cloudformation/cloudformation-cli/pull/644)